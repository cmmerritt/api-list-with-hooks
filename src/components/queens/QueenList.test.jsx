import React from 'react';
import { render, cleanup } from '@testing-library/react';
import QueenList from './QueenList';
import { MemoryRouter } from 'react-router-dom';

const TestQueenList = [
  {
    name: 'RuPaul',
    image: 'rupaul.jpg',
    quote: 'Shante, you stay.'
  },
  {
    name: 'Darcy',
    image: 'darcy.jpg',
    quote: 'Sashay away.'
  }
];

describe('QueenList component', () => {
  afterEach(() => cleanup());
  it('renders QueenList', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <QueenList
          queens={TestQueenList} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
