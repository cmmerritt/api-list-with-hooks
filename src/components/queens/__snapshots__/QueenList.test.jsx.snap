// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueenList component renders QueenList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <figure>
        <a
          href="/queen/undefined"
        >
          <img
            alt="RuPaul"
            height="200px"
          />
          <figcaption>
            <p>
              RuPaul
            </p>
            <p>
              "Shante, you stay."
            </p>
          </figcaption>
        </a>
      </figure>
    </li>
    <li>
      <figure>
        <a
          href="/queen/undefined"
        >
          <img
            alt="Darcy"
            height="200px"
          />
          <figcaption>
            <p>
              Darcy
            </p>
            <p>
              "Sashay away."
            </p>
          </figcaption>
        </a>
      </figure>
    </li>
  </ul>
</DocumentFragment>
`;
